plugins/ftp/vsftpd

```diff
@@ -34,18 +62,24 @@ ftp_su=U
 ftp_fu=U
 ftp_sd=U
 ftp_fd=U
+ftp_sr=U
+ftp_fr=U
 ftp_sde=U
 ftp_fde=U
 
-ftp_c=`grep "CONNECT" ${LOGFILE} | wc -l`
-ftp_sl=`grep "OK LOGIN" ${LOGFILE} | wc -l`
-ftp_fl=`grep "FAIL LOGIN" ${LOGFILE} | wc -l`
-ftp_su=`grep "OK UPLOAD" ${LOGFILE} | wc -l`
-ftp_fu=`grep "FAIL UPLOAD" ${LOGFILE} | wc -l`
-ftp_sd=`grep "OK DOWNLOAD" ${LOGFILE} |wc -l`
-ftp_fd=`grep "FAIL DOWNLOAD" ${LOGFILE} | wc -l`
-ftp_sde=`grep "OK DELETE" ${LOGFILE} |wc -l`
-ftp_fde=`grep "FAIL DELETE" ${LOGFILE} | wc -l`
+if [ -f "${logfile}" ]; then
+	ftp_c=$(grep -c "CONNECT" "${logfile}")
+	ftp_sl=$(grep -c "OK LOGIN" "${logfile}")
+	ftp_fl=$(grep -c "FAIL LOGIN" "${logfile}")
+	ftp_su=$(grep -c "OK UPLOAD" "${logfile}")
+	ftp_fu=$(grep -c "FAIL UPLOAD" "${logfile}")
+	ftp_sd=$(grep -c "OK DOWNLOAD" "${logfile}")
+	ftp_fd=$(grep -c "FAIL DOWNLOAD" "${logfile}")
+	ftp_sr=$(grep -c "OK RENAME" "${logfile}")
+	ftp_fr=$(grep -c "FAIL RENAME" "${logfile}")
+	ftp_sde=$(grep -c "OK DELETE" "${logfile}")
+	ftp_fde=$(grep -c "FAIL DELETE" "${logfile}")
+fi
```

Comment:
I think "0" is a better value for known events which did not happen.

To use less resources (and initialize everything to "0", you could also call "awk | sort" once, instead of many instances of "grep".  Example:

    awk '
    BEGIN {
    	count["ftp_c"] = 0
    	count["ftp_sl"] = 0
    	count["ftp_fl"] = 0
    	count["ftp_su"] = 0
    	count["ftp_fu"] = 0
    	count["ftp_sd"] = 0
    	count["ftp_fd"] = 0
    	count["ftp_sde"] = 0
    	count["ftp_fde"] = 0
    }
    /CONNECT/       { count["ftp_c"]++   }
    /OK LOGIN/      { count["ftp_sl"]++  }
    /FAIL LOGIN/    { count["ftp_fl"]++  }
    /OK UPLOAD/     { count["ftp_su"]++  }
    /FAIL UPLOAD/   { count["ftp_fu"]++  }
    /OK DOWNLOAD/   { count["ftp_sd"]++  }
    /FAIL DOWNLOAD/ { count["ftp_fd"]++  }
    /OK DELETE/     { count["ftp_sde"]++ }
    /FAIL DELETE/   { count["ftp_fde"]++ }
    
    END {
    	for (sort label in count)
    		printf "%s.label %d\n", label, count[label]
    }
    ' $logfile | sort